.github/workflows/main.yml

```diff
@@ -2,132 +2,113 @@ name: CI
+
+      - name: npm ci
+        working-directory: angular
+        run: |
+          touch src/environments/jwt.js
+          npm ci
+
+      - name: Unit Tests
+        working-directory: angular
+        run: npm run test:ci
 
   Angular-Linting:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Setup Node.js for use with actions
-      uses: actions/setup-node@v1
-      with:
-        node-version: 14.x
-
-    - uses: actions/cache@v1
-      with:
-        path: ~/.npm
-        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-        restore-keys: |
-          ${{ runner.os }}-node-
-
-    - name: npm ci
-      run: |
-        cd angular
-        npm ci
-    - name: Linting
-      run: |
-        cd angular
-        npm run lint
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js for use with actions
+        uses: actions/setup-node@v4
+        with:
+          node-version: 14
+          cache: npm
+          cache-dependency-path: angular/package-lock.json
+
+      - name: npm ci
+        working-directory: angular
+        run: npm ci
+
+      - name: Linting
+        working-directory: angular
+        run: npm run lint
+
   React-Unit-Tests:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - name: Setup Node.js for use with actions
-        uses: actions/setup-node@v1
-        with:
-          node-version: 20.10.x
+      - uses: actions/checkout@v4
 
-      - uses: actions/cache@v1
+      - name: Setup Node.js for use with actions
+        uses: actions/setup-node@v4
         with:
-          path: ~/.npm
-          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-node-
+          node-version: 20
+          cache: npm
+          cache-dependency-path: react/package-lock.json
 
       - name: npm ci
-        run: |
-          cd react
-          npm ci
+        working-directory: react
+        run: npm ci
+
       - name: Unit Tests
+        working-directory: react
         run: |
-          cd react
           cp src/secret_local.example.ts src/secret_local.ts
           npm run test
+
   React-Linting:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Setup Node.js for use with actions
-      uses: actions/setup-node@v1
-      with:
-        node-version: 20.10.x
-
-    - uses: actions/cache@v1
-      with:
-        path: ~/.npm
-        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-        restore-keys: |
-          ${{ runner.os }}-node-
-
-    - name: npm ci
-      run: |
-        cd react
-        npm ci
-    - name: Linting
-      run: |
-        cd react
-        npm run lint
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js for use with actions
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
```

Comment:
should we match our Dockerfile? or better to test latest version?

maybe we bump Dockerfile?

Reply:
Ah, you mean 20.10.x instead of 20.x? That makes sense. We can also bump the dockerfile

Reply:
Actually I think just using active LTS makes sense, but I bumped it to v22 (LTS)